github/branches/validate_compliance.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+from typing import List
+
+from .retrieve_github_data import RepoBranchProtections, BranchProtectionRule
+from ..helpers import Criteria
+
+
+# define the criteria we care about. Identify each critera with a string that will
+# appear in the results.
+required_criteria = [
+    Criteria("SOGH001", "rules", "active rules"),
+    Criteria("SOGH001b", "admins", "admins restricted"),
+]
+optional_criteria = [
+    Criteria("SOGH001c", "commiters", "allowed commiters configured"),
+    # "commit signing",  # may not be knowable
+]
+warning_criteria = [
+    Criteria("SOGH001d", "conflicts", "Conflict in Protection Rules"),
+]
+
+
+def find_applicable_rules(
+    data: RepoBranchProtections, branch: str
+) -> List[BranchProtectionRule]:
+    result = []
+    for rule in data.protection_rules:
+        for ref_name in rule.matching_branches:
+            if branch == ref_name.branch_name:
+                result.append(rule)
+                break
+    return result
+
+
+def meets_criteria(protections: List[BranchProtectionRule], criteria: Criteria) -> bool:
+    met = True
+    # ugly implementation for now
+    if criteria.slug == "rules":
+        met = len(protections) > 0
+    elif criteria.slug == "admins":
+        met = all(r.is_admin_enforced for r in protections)
+    elif criteria.slug == "commiters":
+        met = all(r.push_actor_count > 0 for r in protections)
+    elif criteria.slug == "conflicts":
+        met = all(r.rule_conflict_count == 0 for r in protections)
+    else:
+        met = False
+    return met
+
+
+def validate_branch_protections(
```

Comment:
Would love to see a doctest for this func

Reply:
Coming soon in (another) branch near you.  This is the part I may have to refactor heavily as I automate issue generation.

Leaving this open for now, I'll (hopefully) be able to pull in a solution.